lib/std/child_process.zig

```diff
@@ -194,6 +194,25 @@ pub const ChildProcess = struct {
         stderr: []u8,
     };
 
+    pub fn collectOutput(
```

Comment:
Since this method is now public, it would be nice to have a doc comment that 
- 1. it collects stdout and stderr, 
- 2. caller owns stdout and stderr memory, 
- 3. it is non-blocking.

Not sure, if a  hint to use `wait()` for blocking is also nice.

Quick reminder for otther reviewers, that this has currently no unit tests unless #11736 is merged.

Reply:
> caller owns stdout and stderr memory,


This is not usually documented when passing pointer to `ArrayList` (see `Reader.readAllArrayList`). This make sense. Taking mutable pointer mean we are going to mutate the datastructure, which would mean inserting/deleting elements from it. This still implies that the structure is owned by the caller. After all, they initialized it.